sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -145,3 +153,24 @@ def test_transpose_power():
 
     assert ((D*C)**-5).T**-5 == ((D*C)**25).T
     assert (((D*C)**l).T**k).T == (D*C)**(l*k)
+
+
+def test_Inverse():
+    assert Inverse(MatPow(C, 0)).doit() == Identity(n)
+    assert Inverse(MatPow(C, 1)).doit() == Inverse(C)
+    assert Inverse(MatPow(C, 2)).doit() == MatPow(C, -2)
+    assert Inverse(MatPow(C, -1)).doit() == C
+
+    assert MatPow(Inverse(C), 0).doit() == Identity(n)
+    assert MatPow(Inverse(C), 1).doit() == Inverse(C)
+    assert MatPow(Inverse(C), 2).doit() == MatPow(C, -2)
+    assert MatPow(Inverse(C), -1).doit() == C
+
+
+def test_combine_powers():
+    assert (C ** 1) ** 1 == C
+    assert (C ** 2) ** 3 == MatPow(C, 6)
+    assert (C ** -2) ** -3 == MatPow(C, 6)
```

Comment:
It would also be good to have tests showing that the equivalents of these with MatPow do not evaluate to anything e.g.:
```
from sympy.core.expr import unchanged

C2 = MatPow(C, 2)
assert unchanged(MatPow, C, 0)
assert unchanged(MatPow, C, 1)
assert unchanged(MatPow, C, -1)
assert unchanged(MatPow, C2, 2)
```

Reply:
It seems there are no tests at all like that in the `MatPow` tests. The idea of `unchanged` is that it asserts that the expression does not evaluate or simplify in any way which is the intention here e.g.:
```julia
In [7]: MatPow(Identity(2), 2)                                                                                                    
Out[7]: 
   2
(𝕀) 

In [8]: MatPow(ZeroMatrix(2, 2), 2)                                                                                               
Out[8]: 
   2
(𝟘) 
```